OpenRA.Mods.Common/Traits/World/ResourceLayer.cs

```diff
@@ -35,6 +35,8 @@ public class ResourceLayer : IRenderOverlay, IWorldLoaded, ITickRender, INotifyA
 		protected readonly CellLayer<CellContents> Content;
 		protected readonly CellLayer<CellContents> RenderContent;
 
+		public bool IsEmpty { get { return !Content.Any(); } }
```

Comment:
A CellLayer is just the set of cells on the map, so it's not ever going to be empty. Did you intend this to check if there were any resources on the map?